core/src/main/scala/org/apache/spark/network/BlockTransferService.scala

```diff
@@ -99,7 +106,8 @@ abstract class BlockTransferService extends ShuffleClient with Closeable with Lo
         }
       })
 
-    Await.result(result.future, Duration.Inf)
+    val fetchTimeout = new FiniteDuration(timeout, TimeUnit.SECONDS)
+    Await.result(result.future, fetchTimeout)
```

Comment:
You can import `scala.concurrent.duration._` and `import scala.language.postfixOps`, then use `timeout seconds` directly.